src/Adapter/Chain/CachePoolChain.php

```diff
@@ -24,16 +24,32 @@ class CachePoolChain implements CacheItemPoolInterface, TaggablePoolInterface
      * @type CacheItemPoolInterface[]
      */
     private $pools;
+    /** @type bool */
+    private $skipOnFailure;
 
     /**
      * @param array $pools
+     * @param bool  $skipOnFailure
      */
-    public function __construct(array $pools)
+    public function __construct(array $pools, $skipOnFailure = false)
     {
-        if (empty($pools)) {
+        $this->pools         = $pools;
+        $this->skipOnFailure = $skipOnFailure;
+        if (empty($this->pools)) {
             throw new \LogicException('At least one pool is required for the chain.');
         }
-        $this->pools = $pools;
+    }
+
+    /**
+     * @return array|\Psr\Cache\CacheItemPoolInterface[]
+     */
+    public function getPools()
+    {
+        if (empty($this->pools)) {
```

Comment:
If a cache pool fails during a request, Is it a fatal issue that would make us not want to check it again later in the request?